### PR TITLE
Devtooling-394: Refactored outbound settings

### PR DESCRIPTION
**Pull request by BrianMoyles**

### CODE REVIEW

File: genesyscloud/outbound_settings/genesyscloud_outbound_settings_init_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	resourceMapMutex sync.RWMutex
+}
+
+// registerTestResources registers all resources used in the tests
+func (r *registerTestInstance) registerTestResources() {
+	r.resourceMapMutex.Lock()
+	defer r.resourceMapMutex.Unlock()
+
+	providerResources[resourceName] = ResourceOutboundSettings()
+	// TODO: Add references
```
Comment (dginty4) [inline genesyscloud/outbound_settings/genesyscloud_outbound_settings_init_test.go:30]: You can remove this TODO

File: genesyscloud/outbound_settings/resource_genesyscloud_outbound_settings.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+func getAllOutboundSettings(_ context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	resources["0"] = &resourceExporter.ResourceMeta{Name: "outbound_settings"}
+	return resources, nil
+}
+
+// createOutboundSettings is used by the outbound_settings resource to create Genesys cloud outbound settings
+func createOutboundSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("Creating Outbound Setting")
```
Comment (HemanthDogiparthi12) [inline genesyscloud/outbound_settings/resource_genesyscloud_outbound_settings.go:28]: add more meaninful log, else you can remove this as well

File: genesyscloud/outbound_settings/resource_genesyscloud_outbound_settings.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	d.SetId("settings")
```
Comment (HemanthDogiparthi12) [inline genesyscloud/outbound_settings/resource_genesyscloud_outbound_settings.go:29]: why is this set to hardcoded value

File: genesyscloud/outbound_settings/resource_genesyscloud_outbound_settings.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+				d.Set("compliance_abandon_rate_denominator", *settings.ComplianceAbandonRateDenominator)
+			} else {
+				d.Set("compliance_abandon_rate_denominator", nil)
+			}
+		}
+
+		if len(automaticTimeZoneMapping) > 0 {
+			d.Set("automatic_time_zone_mapping", flattenOutboundSettingsAutomaticTimeZoneMapping(*settings.AutomaticTimeZoneMapping, automaticTimeZoneMapping))
+		}
+		log.Printf("Read Outbound Setting")
```
Comment (HemanthDogiparthi12) [inline genesyscloud/outbound_settings/resource_genesyscloud_outbound_settings.go:93]: same  add more context to the log.  d.id() may be

File: genesyscloud/outbound_settings/resource_genesyscloud_outbound_settings_schema.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+1.  The registration code that registers the Datasource, Resource and Exporter for the package.
+2.  The resource schema definitions for the outbound_settings resource.
+3.  The datasource schema definitions for the outbound_settings datasource.
+4.  The resource exporter configuration for the outbound_settings exporter.
+*/
+const resourceName = "genesyscloud_outbound_settings"
+
+// SetRegistrar registers all the resources, datasources and exporters in the package
+func SetRegistrar(l registrar.Registrar) {
+	l.RegisterResource(resourceName, ResourceOutboundSettings())
```
Comment (HemanthDogiparthi12) [inline genesyscloud/outbound_settings/resource_genesyscloud_outbound_settings_schema.go:24]: is there no datasource for this one ?

**Review by HemanthDogiparthi12 — CHANGES_REQUESTED** [review]

**Comment by dginty4** [comment]
@HemanthDogiparthi12 the answer to most of your comments is due to the fact that outbound settings does not have an id or name property. There is only one outbound settings object in the org and it comes with the org, the cx as code resource just allows the user to configure the settings through terraform

**Review by dginty4 — APPROVED** [review]

**Review by HemanthDogiparthi12 — APPROVED** [review]

**Pull request merged**